File: 202-happy-number/happy-number.java
class Solution {

    public int next(int num){
        int sum = 0;
        while(num > 0){
            int digit = num%10;
            sum += digit*digit;
            num = num/10;
        }
        return sum;
    }

    public boolean isHappy(int n) {
        int slow = n;
        int fast = next(n);

        while(slow != fast){
            slow = next(slow);
            fast = next(next(fast));
        }
        if(slow == 1) return true;
        return false;
    }
}

File: 202-happy-number/happy-number.py
class Solution:
    def get_next(self,num):
        result=0
        while num>0:
            digit = num%10
            result+=digit*digit
            num=num//10
        return result

    def isHappy(self, n: int) -> bool:

        slow=n
        fast=n
        while True:
            slow=self.get_next(slow)
            fast=self.get_next(self.get_next(fast))
            if slow==fast:
                break
        return slow==1
